app/src/picovr/cpp/DeviceDelegatePicoVR.cpp

```diff
@@ -373,6 +401,23 @@ DeviceDelegatePicoVR::StartFrame() {
 
   m.cameras[0]->SetHeadTransform(head);
   m.cameras[1]->SetHeadTransform(head);
+
+
+  // Update te gaze mode state based on controllers availability
+  m.isInGazeMode = true;
+  for (int32_t i = 0; i < m.controllers.size(); ++i) {
+    if (i != m.gazeIndex) {
+      m.isInGazeMode &= !m.controllers[i].enabled;
```

Comment:
I'm not sure I like using a bitwise and operator with booleans. Maybe:
```
  if (!m.controllers[i].enabled) {
    m.isInGazeMode = false;
    break;
  }
```

Reply:
Updated

Reply:
Updated